src/librustc_typeck/check/mod.rs

```diff
@@ -3344,6 +3344,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                 }
                             };
                     }
+                    ty::Array(ty, _) if ty.is_numeric() => {
+                        let base = self.tcx.hir.node_to_pretty_string(base.id);
+                        let msg = format!("attempting to use tuple indexing on an array; try");
```

Comment:
Let's change the wording to something along the lines of `instead of using tuple indexing use array indexing`.